astlib/migrate_501_500.ml

```diff
@@ -787,13 +788,32 @@ and copy_module_expr_desc :
       Ast_500.Parsetree.Pmod_functor
         (copy_functor_parameter x0, copy_module_expr x1)
   | Ast_501.Parsetree.Pmod_apply (x0, x1) ->
-      Ast_500.Parsetree.Pmod_apply (copy_module_expr x0, copy_module_expr x1)
+      let x1 = copy_module_expr x1 in
+      let x1 =
+        match x1.pmod_desc with
+        | Pmod_structure [] ->
+            let pmod_attributes =
+              {
+                Ast_500.Parsetree.attr_name =
+                  {
+                    txt = "ppxlib.migration.keep_structure";
+                    loc = { x1.pmod_loc with loc_ghost = true };
+                  };
+                attr_payload = Ast_500.Parsetree.PStr [];
+                attr_loc = Location.none;
+              }
+              :: x1.pmod_attributes
+            in
+            { x1 with pmod_attributes }
+        | _ -> x1
+      in
+      Ast_500.Parsetree.Pmod_apply (copy_module_expr x0, x1)
   | Ast_501.Parsetree.Pmod_apply_unit x0 ->
       let empty_struct =
         Ast_500.Parsetree.
           {
             pmod_desc = Pmod_structure [];
-            pmod_loc = Location.none;
+            pmod_loc = loc;
```

Comment:
Shouldn't this be a ghost location?

Reply:
Yes, I guess it should be...
Currently it is not really used (the upward migration will not use it) but when the ppxlib AST is bumped to 5.1, it will be used.
I think I don't know sure enough how tooling treats ghost locations.

Reply:
Actually, making it ghost creates one more divergence in the `reverse_migrations.t` test. So I guess it makes sense to leave it not ghost.

Reply:
> I think I don't know sure enough how tooling treats ghost locations.

:heavy_plus_sign:

Reply:
Some day (:tm:) I'll have a concrete look at what Merlin does.